app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java

```diff
@@ -899,11 +899,8 @@ private void handleSendImage(Intent intent) {
         Uri selectedImage = null;
         ArrayList<Uri> selectedImagesArray = new ArrayList<>();
         selectedImage = intent.getParcelableExtra(Intent.EXTRA_STREAM);
-        boolean isBarCodePresent = false;
-        if (selectedImage != null) {
-            selectedImagesArray.add(selectedImage);
-            chooseDialog(selectedImagesArray);
-        }
+        selectedImagesArray.add(selectedImage);
```

Comment:
Why did you remove the check if (selectedImage != null) {....
Is it not usefull ?

Reply:
Could leave it in, just removed it as running chooseDialog() then runs detectBarCodeInImage(), and the first line of that is the added 'if (uri == null)' so it checks for null anyway

Maybe I should leave it in to check for null before so it doesn't waste time getting to the new null check

Reply:
if "selectedImage" is null, chooseDialog could be launched with a null value, then `product_images.setAdapter(new PhotosAdapter( uri));` (line 983) will, then `PhotosAdapter#onBindViewHolder`
But I'm not sure it has an impact....

Reply:
Ah I see why that would be a problem - I will add that back in, but if one of the Uris in the ArrayList is null (because it's invalid) after `handleSendMultipleImages` is run, that line (`MainActivity#983`) will create a `PhotosAdapter` with an ArrayList containing a null value - if `PhotosAdapter#onBindViewHolder` is called then will it also cause a NullPointerException at `PhotosAdapter#40` (`load(mPhotos.get(position)`).

To solve all of these problems, I could re-add the null check in `handleSendImage`  and at the start of `handleSendMultipleImages` iterate over the ArrayList to remove null values; this would mean the new `if (uri == null) return false` line can be removed

What do you think?

Reply:
👍  Ok with your last suggestion 
Thanks